### ISSUE TITLE
breeze, r.js, and almond

### ISSUE THREAD

**Issue by mattrandle** [Breeze/breeze.js#54]
As it stands, breeze doesnt work with almond and r.js without modifications.

There are two issues,
1. There are multiple anonymous defines in the breeze source file so r.js cant add module ids during the optimization process.  The additional defines are for the jquery ajax adapter, the web api data store, etc. Im struggling to see why the adapters have defines at all - in the breeze.js file, the breeze variable exists in the closure so in this code (copied from the ajax adapter),

if (breeze) {
            factory(breeze);
} else if (typeof require === "function" && typeof exports === "object" && typeof module === "object") {
            // CommonJS or Node: hard-coded dependency on "breeze"
            factory(require("breeze"));
} else if (typeof define === "function" && define["amd"]) {
            // AMD anonymous module with hard-coded dependency on "breeze"
            define(["breeze"], factory);
}

'if (breeze)' always evaluates to true and the other code never runs.

Also, the require documentation states you should only have a single module per file.

2 The __requireLibCore function does a check for require using code,

var r = window.require;
if (r) ....... 

Unlike require.js, almond doesnt insert anything into the global scope so window.require is always null. However, the require function is available in the closure (assuming breeze is bundled into the same file) so this works,

var r = window.require || require;

---

In summary, I think that making breeze run ootb with r.js and almond is a desirable feature and is not too hard (granted ive only done 15 minutes of analysis).

I know there are other ways to bundle breeze, but it would be nice if you could just use breeze with r.js if thats the best solution for your project.

I dont claim to be a require, r.js, or breeze expert so I might have missed something obvious you can do with the config to make this work.
